### PR TITLE
FCS pi0 finder for Ecal by Xilin

### PR DESCRIPTION
**Pull request by akioogawa**
This will be needed for FCS pi0 peaks to be part of FastOffline.

### CODE REVIEW

File: StRoot/StSpinPool/StFcsPi0FinderForEcal/StFcsPi0FinderForEcal.cxx [hunk 1]
```diff
@@ -0,0 +1,546 @@
...
+		return kStOK;
+
+}
+
+
+	//-----------------------
+	Int_t StFcsPi0FinderForEcal::Finish()
+	{
+		const char* fn=filename.c_str();
+		if (filename.length()==0) return kStOk;
```
Comment (genevb) [inline StRoot/StSpinPool/StFcsPi0FinderForEcal/StFcsPi0FinderForEcal.cxx]: Not too important, but the test for filename.length() can come before the assignment of fn.

**Review by zlchang — APPROVED** [review]

File: StRoot/StSpinPool/StFcsPi0FinderForEcal/StFcsPi0FinderForEcal.cxx [hunk 2]
```diff
@@ -0,0 +1,507 @@
...
+StFcsPi0FinderForEcal::~StFcsPi0FinderForEcal() {}
+
+//-----------------------
+Int_t StFcsPi0FinderForEcal::Init() {
+   mFcsDb = static_cast<StFcsDb*>(GetDataSet("fcsDb"));
+   mFcsDb->setDbAccess(0);
+   if (!mFcsDb) {
+      LOG_ERROR << "StFcsEventDisplay::InitRun Failed to get StFcsDbMaker" << endm;
+      return kStFatal;
+   }
```
Comment (veprbl) [inline StRoot/StSpinPool/StFcsPi0FinderForEcal/StFcsPi0FinderForEcal.cxx:53]: Makes no sense to first dereference a pointer and then check it for NULL.

File: StRoot/StSpinPool/StFcsPi0FinderForEcal/StFcsPi0FinderForEcal.cxx [hunk 3]
```diff
@@ -0,0 +1,507 @@
...
+   h2_point_dgg_vs_E1pE2->SetYTitle("d_{gg} [cm]");
+
+   return kStOK;
+}
+
+//-----------------------
+Int_t StFcsPi0FinderForEcal::Finish() {
+   if (filename.length() == 0) return kStOk;
+   const char* fn = filename.c_str();
+   TFile* MyFile = new TFile(fn, "RECREATE");
```
Comment (veprbl) [inline StRoot/StSpinPool/StFcsPi0FinderForEcal/StFcsPi0FinderForEcal.cxx:163]: This leaks memory. Consider switching to
```suggestion
   TFile MyFile(fn, "RECREATE");
```
or using `std::unique_ptr<TFile>`

File: StRoot/StSpinPool/StFcsPi0FinderForEcal/StFcsPi0FinderForEcal.h [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+   int bins = 150;
+   string filename;
+   float m_low = 0;
+   float m_up = 0.4;
+   float E_up = 10;
+   float E_low = 8;
+   float E_min = 1;
+
+#ifndef SKIPDefImp
+   ClassDef(StFcsPi0FinderForEcal, 0)
```
Comment (veprbl) [inline StRoot/StSpinPool/StFcsPi0FinderForEcal/StFcsPi0FinderForEcal.h:94]: Not sure why this is behind an ifdef, our https://github.com/star-bnl/star-sw/blob/main/mgr/RootCint.pl will still see this and try to include it in a generated LinkDef.

**Review by veprbl — APPROVED** [review]
I took a liberty to reformat the source (feel free to revert). Don't see any logical issues, left some comments about technical ones.

**Comment by plexoos** [comment]
@veprbl raised very reasonable concerns about the code. Too bad they were ignored...

File: StRoot/StSpinPool/StFcsPi0FinderForEcal/StFcsPi0FinderForEcal.cxx (hunk 2 above)
Comment (akioogawa) [reply StRoot/StSpinPool/StFcsPi0FinderForEcal/StFcsPi0FinderForEcal.cxx:53]: Agreed

File: StRoot/StSpinPool/StFcsPi0FinderForEcal/StFcsPi0FinderForEcal.cxx (hunk 3 above)
Comment (akioogawa) [reply StRoot/StSpinPool/StFcsPi0FinderForEcal/StFcsPi0FinderForEcal.cxx:163]: Yes. But it is in Finish() not in Make(). Still good idea.

File: StRoot/StSpinPool/StFcsPi0FinderForEcal/StFcsPi0FinderForEcal.h (hunk 4 above)
Comment (akioogawa) [reply StRoot/StSpinPool/StFcsPi0FinderForEcal/StFcsPi0FinderForEcal.h:94]: This is what Gene did as a trick to include this maker fastoffline run off from STARDEV while this Maker is not in DEV (StSpinPool). He said this trick works.

**Comment by genevb** [comment]
I merged this PR to get QA moving. I don't disagree that @veprbl 's points are reasonable and the authors can & should address them promptly (soon), but I don't perceive those points as more critical than QA'ing data we are taking at the experiment _right_now_ (this PR already took a week). Given @veprbl 's approval of this PR, perhaps he felt the same.

**Comment by akioogawa** [comment]
@plexoos  @veprbl  @genevb Sorry for not attending on those comments. Crazy days. Now this is merged and closed, what do I do? Should/CanI re-open, or create whole new PR?

**Comment by veprbl** [comment]
@akioogawa New PR is the way to make further changes. Thanks!

**Comment by akioogawa** [comment]
See PR #271

**Pull request merged**